File: library_management/library_management/report/library_member_report/library_member_report.py
# Copyright (c) 2026, Saravanan M K and contributors
# For license information, please see license.txt

import frappe


def execute(filters=None):
    columns = ["Member Name:Data:200", "Membership Type:Data:150", "Active Member:Check:120"]
    data = frappe.get_all("Library Member",fields=[ "member_name", "membership_type", "active_member"])
    return columns, data

File: library_management/library_management/report/library_member_report/library_member_report.js
// Copyright (c) 2026, Saravanan M K and contributors
// For license information, please see license.txt

frappe.query_reports["Library Member Report"] = {
    filters: [
        {
            fieldname: "membership_type", label: "Membership Type", fieldtype: "Select", options: "\nGold\nSilver\nPlatinum"
        },
        {
            fieldname: "active_member", label: "Active Member", fieldtype: "Check"
        }
    ]
};
